tests/test_architect.py

```diff
@@ -635,3 +636,82 @@ def test_test_matrix(self):
                 with open(matrix_filename, 'r') as f:
                     reader = csv.reader(f)
                     assert(len([row for row in reader]) == 13)
+
+    def test_replace(self):
+        with testing.postgresql.Postgresql() as postgresql:
+            # create an engine and generate a table with fake feature data
+            engine = create_engine(postgresql.url())
+            create_features_and_labels_schemas(engine, features_tables, labels)
+
+            dates = [datetime.datetime(2016, 1, 1, 0, 0),
+                     datetime.datetime(2016, 2, 1, 0, 0),
+                     datetime.datetime(2016, 3, 1, 0, 0)]
+
+            with TemporaryDirectory() as temp_dir:
+                matrix_maker = Architect(
+                    beginning_of_time = datetime.datetime(2010, 1, 1, 0, 0),
+                    label_names = ['booking'],
+                    label_types = ['binary'],
+                    db_config = db_config,
+                    matrix_directory = temp_dir,
+                    user_metadata = {},
+                    engine = engine,
+                    replace=False
+                )
+
+                matrix_dates = {
+                    'matrix_start_time': datetime.datetime(2016, 1, 1, 0, 0),
+                    'matrix_end_time': datetime.datetime(2016, 3, 1, 0, 0),
+                    'as_of_times': dates
+                }
+                feature_dictionary = {
+                    'features0': ['f1', 'f2'],
+                    'features1': ['f1', 'f2'],
+                }
+
+                uuid = '1234'
```

Comment:
the temporary csv branch will require a read uuid for tests (because timechop will delete its csv). i can fix this in that pr.